### DISCUSSION TITLE
Original CAD Files

### DISCUSSION THREAD

**Discussion by ghost** [iliasam/OpenSimpleLidar#13] [Q&A] [answered]
I'm working on assembling the laser module, but it's getting stuck in the holder and won't fit properly. It seems like this might be due to the round holes being converted into polygons in the STL file. I tried sanding down the 3D-printed part, but that makes it hard to keep the laser module aligned vertically.

May you offer the original CAD files so I can make some adjustments? Thanks！
![img](https://github.com/user-attachments/assets/293f5337-c2cc-42e5-8dd7-17cbf01cffbf)

**Comment by iliasam** [comment, marked as answer]
Sorry, I draw parts in non-pupular software.
Maybe it is possible to do conversion, but I need to know need file type.
I have published STEP files, maybe it can help you.

Can you show photo - what is wrong with your printed pates?

**Comment by iliasam** [comment]
Just notice - laser is placed inside the "laser holder" at certain angle, laser beam is NOT perpendicular to the PCB:
https://raw.githubusercontent.com/iliasam/OpenSimpleLidar/refs/heads/master/Misc/optics1.png

**Reply by ghost** [reply]
Okay, I'll keep an eye on it. Thanks！

**Comment by chrisalbertson** [comment]
> On Dec 19, 2024, at 7:46 AM, iliasam ***@***.***> wrote:
> 
> 
> I have published STEP files, maybe it can help you.
> 

If you have spublished step files, you have done what the OP asked for,  Those should be editable in any CAD software.

The problem is likey NOT with the design, but that pprinters are not so exact, I aliitle bit of over extrusion and any hole will become smaller.   Usually you can fix this problem in the slicer

As for holes being too small, a drill can fix that.

**Reply by ghost** [reply]
Indeed, the diameter in the original stl file is larger than 6mm, which should have been just enough to fit the laser module. 
I do not have a 3D printer. I opted for SLA 3D printing services from an online provider. They gave me feedback that the resin material tends to shrink and distort, and suggested that I try a hot air gun to soften the distorted area to correct it. Anyway, thanks for your help!

**Comment by chrisalbertson** [comment]
I do a lot of printing in PLA and lately ABS.    You really need to think of 2D printed parts as cast metal parts, not as CNC machined parts.  Everyone who builds with 3D printers keeps tools like rat-tail and flat files and a few different gits of sand paper in his kit.  I happen to have some machine tools and many times I’ll chuck a printed part in t a lathe to make the exact.
